posts/2020-08-27-Rust-1.46.0.md

```diff
@@ -0,0 +1,123 @@
+---
+layout: post
+title: "Announcing Rust 1.46.0"
+author: The Rust Release Team
+release: true
+---
+
+The Rust team is happy to announce a new version of Rust, 1.46.0. Rust is a
+programming language that is empowering everyone to build reliable and
+efficient software.
+
+If you have a previous version of Rust installed via rustup, getting Rust
+1.46.0 is as easy as:
+
+```console
+rustup update stable
+```
+
+If you don't have it already, you can [get `rustup`][install] from the
+appropriate page on our website, and check out the [detailed release notes for
+1.46.0][notes] on GitHub.
+
+[install]: https://www.rust-lang.org/install.html
+[notes]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1460-2020-08-27
+
+## What's in 1.46.0 stable
+
+This release is on the smaller side, with a number of improvements to `const
+fn`, two new standard library APIs, and one feature useful for library
+authors. See the [detailed release notes][notes] to learn about other changes
+not covered by this post.
+
+### `#[track_caller]`
+
+Back in March, the release of Rust 1.42 brought about [better error messages when unwrap would panic][better-errors]. At the time, we mentioned that the way
+this was implemented was not yet stable. Rust 1.46 stabilizes this feature.
+
+[better-errors]: https://blog.rust-lang.org/2020/03/12/Rust-1.42.html#useful-line-numbers-in-option-and-result-panic-messages
+
+This attribute is called `#[track_caller]`, which was originally proposed
+in [RFC 2091][rfc-2091] way back in July of 2017! If you're writing a function
+like unwrap, where you want errors to report to the caller, you can put this
+annotation on your functions, and the default panic formatter will use it to
+print its error message. For example, here is `unwrap` previously:
+
+```rust
+pub fn unwrap(self) -> T {
+    match self {
+        Some(val) => val,
+        None => panic!("called `Option::unwrap()` on a `None` value"),
+    }
+}
+```
+
+It now looks like this:
+
+```rust
+#[track_caller]
+pub fn unwrap(self) -> T {
+    match self {
+        Some(val) => val,
+        None => panic!("called `Option::unwrap()` on a `None` value"),
+    }
+}
+```
+
+That's it!
```

Comment:
...though an example that uses `panic::Location` might be helpful.

Reply:
I wanted to emphasize how simple this is to use, but if you wanted to come up with an example using Location, I'm not inherently opposed. I didn't show how the output was changed because we did that when it was added to unwrap.